backend/repos/query.go

```diff
@@ -0,0 +1,142 @@
+package repos
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/honeynet/ochi/backend/entities"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/jmoiron/sqlx/reflectx"
+	_ "github.com/mattn/go-sqlite3"
+)
+
+type QueryRepo struct {
+	findByOwnerQuery *sqlx.Stmt
+	deleteQuery      *sqlx.Stmt
+	createQuery      *sqlx.NamedStmt
+	updateQuery      *sqlx.Stmt
+	getByIdQuery     *sqlx.Stmt
+	db               *sqlx.DB
+}
+
+func NewQueryRepo(db *sqlx.DB) (*QueryRepo, error) {
+	r := &QueryRepo{}
+	db.Mapper = reflectx.NewMapperFunc("json", strings.ToLower)
+	_, err := db.Exec(`CREATE TABLE IF NOT EXISTS queries (
+		id TEXT PRIMARY KEY NOT NULL
+		, content TEXT NOT NULL
+		, owner_id TEXT NOT NULL
+		, active INTEGER NOT NULL
+		, description TEXT NOT NULL
+		, CONSTRAINT id_unique UNIQUE (id)
+	)`)
+	if err != nil {
+		return nil, err
+	}
+	r.findByOwnerQuery, err = db.Preparex("SELECT * FROM queries WHERE owner_id=?")
+	if err != nil {
+		return nil, err
+	}
+	r.createQuery, err = db.PrepareNamed("INSERT INTO queries (id, owner_id, content, active, description) VALUES (:id, :owner_id, :content, :active, :description)")
+	if err != nil {
+		return nil, err
+	}
+	r.updateQuery, err = db.Preparex("UPDATE queries SET content=?, active=?, description=? WHERE id=?")
+	if err != nil {
+		return nil, err
+	}
+	r.getByIdQuery, err = db.Preparex("SELECT * FROM queries WHERE id=?")
+	if err != nil {
+		return nil, err
+	}
+	r.deleteQuery, err = db.Preparex("DELETE FROM queries WHERE id=?")
+	if err != nil {
+		return nil, err
+	}
+	return r, nil
+}
+
+// Get a user
+func (r *QueryRepo) FindByOwnerId(ownerId string) ([]entities.Query, error) {
+	qs := []entities.Query{}
+	err := r.findByOwnerQuery.Select(&qs, ownerId)
+	return qs, err
+}
+
+// Create a query
+func (r *QueryRepo) Create(owner_id, content, description string, active bool) (entities.Query, error) {
+	id, err := uuid.NewRandom()
+	if err != nil {
+		return entities.Query{}, err
+	}
+	q := entities.Query{ID: id.String(), Content: content, OwnerID: owner_id, Active: active, Description: description}
```

Comment:
initialize this first so you can return it in error cases so you don't need to re-allocate on every error.

Reply:
Done